### DISCUSSION TITLE
Hetero Temporal Data

### DISCUSSION THREAD

**Discussion by DogitoErgoSum** [pyg-team/pytorch_geometric#4323] [Ideas]
Is there a data class like TemporalData but for different node types?

**Comment by rusty1s** [comment]
Currently, `TemporalData` represents a list of events given by the tuple (`src_node_idx, dst_node_idx, timestamp, msg_feature)`. As such, indices of nodes can represent single types or multiple types.

In particular, our [`TGN` example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/tgn.py) makes use of a Wikipedia dataset in which nodes represent both users and pages.

Pinging @otaviocx and @Padarn for additional input.

**Comment by otaviocx** [comment]
Hi @DogitoErgoSum, thanks for asking!

We are currently in the process to improve the support for temporal graphs (see: #3230). But, the `TemporalData` class as-is allows you to add custom keys to represent, for instance, the type of the event (or even the node).

Could you please describe a bit more the scenario you are needing hetero temporal data (which architecture, for instance) for us to understand better the requirements?

**Reply by DogitoErgoSum** [reply]
yes, see #4324 for example. There i show an example of a toy graph where i shall be doing graph classification
